### ISSUE TITLE
Crashes in El Capitan

### ISSUE THREAD

**Issue by kurisubrooks** [alxn1/wjoy#84]
The crash occurs when entering your Password, to allow the app admin privileges. 
It doesn't pop up a window of a crash log or anything, and i'm unsure if it even saves one, so please let me know how I can provide it to you.

**Comment by heitortsergent** [comment]
Does "turn off rootless" like mentioned in #81 help at all @kurisubrooks?

**Comment by alxn1** [comment]
Yes, latest WJoy build can work in El Capitan only if "System Integrity Protection" disabled.

To disable this "feature" of El Capitan:
1. reboot to recovery
2. open terminal
3. enter command 'csrutil disable', press enter
4. reboot to El Captipan

Now "System Integrity Protection" disabled.

And, if WJoy can't pair with wiimotes, try to open System Preferences, Bluetooth settings, and remove all wiimotes from it. Sometimes it's needed 2 or 3 times.

**Comment by kurisubrooks** [comment]
Thanks for your help. Kinda strange how you need to do this though.

**Comment by alxn1** [comment]
:)

**Comment by eighteight** [comment]
I still can't connect wii remote on El Capitan, even after turning off System Integrity Protection.

Would appreciate any suggestions on what to try next.

**Comment by zacharyvincze** [comment]
Eighteight, can you give me some more information on your problem?  What controller are you using, are you pressing the sync button?  I want to help you with your problem.

**Comment by eighteight** [comment]
Hi Zvincze: I am using nintendo wii remote, and I am pressing the sync buttons (1 and 2 or red button next to batteries). However, at this point I decided to switch to a wired solution.

**Comment by zacharyvincze** [comment]
Eighteight, that's fine.  If you ever chose to try and use a Wiimote again, however; try to go into your **Preferences > Bluetooth** and **remove your Wiimote from the devices menu.**  After, try to connect your controller again.  If this doesn't end up working, you can always try to **reinstall WJoy.**  Make sure you **_trash the old WJoy and empty the trash before installing the next one.**_

I wish you the best!

**Comment by chabbster** [comment]
I have the same issue with Mac OS Sierra on MacBook Pro 2015... System Integrity Protection is already disabled - any idea or a fix?

**Comment by thejacer87** [comment]
same issue as @chabbster ... SIP is disabled... still crashes

on macOS sierra

**Comment by koalalorenzo** [comment]
Any news about this? I am affected as well on macOS Sierra!

**Comment by snesgab** [comment]
me too

**Comment by CymanChili** [comment]
I just posted my experience with WJoy [here](https://github.com/JustinBis/wjoy-foohid/issues/1#issuecomment-323188225) - failing to successfully keep WJoy running permanently and the Wii U Pro Controller connected.
If anyone has any advice on how to make it successfully run on macOS 10.12, please share your solution!

**Comment by joehinkle11** [comment]
bump

**Comment by MrDemocracy** [comment]
I managed to make it work on High Sierra.

- Go to `/Applications/WJoy.app/Contents/MacOS`
- Open Terminal
- Type sudo in terminal, and drag `WJoy` into the Terminal window
- Hit return, and enter your password and hit return one more time

**Comment by UncreativeUsername23** [comment]
Still doesn't work, i've tried everything, disabling SIP, reinstalled it 4 times, did the terminal thing yet nothing works, it keeps opening a window saying "Wjoy wants to make changes in your computer" and it asks for my password, when i enter it, wjoy just crashes.

**Comment by MrDemocracy** [comment]
It shouldn't ask you for your password a second time if you've started it with sudo. 
Did you enter you password in the Terminal window, and did you get any error messages when doing so? 
And just to clarify, did you drag the WJoy.app into Terminal, or the "Contents/MacOS"-executable?
Also you can't close the Terminal window while you are using WJoy, as this will also terminate the WJoy process.

**Comment by CymanChili** [comment]
[This](https://github.com/JustinBis/wjoy-foohid/issues/1#issuecomment-336401169) worked (out of the box) for me with High Sierra. SIP enabled.

**Comment by Christian3399** [comment]
I tried to do what MrDemocracy said, and it didn't work the first time. However, I remembered to change the directory in the terminal to my /Applications/WJoy.app/Contents/MacOS directory by using the cd command followed by my full directory clause. Then I ran the program with sudo and now Wjoy is up and running!

**Comment by msmultitool** [comment]
Hi all, i know the thread is quite old, but anyway: 
i'm currently testing to use the Wii Remote as a Midi controller without having to use the Wii Box. The Wii Remote controls a Band hero drumkit (release 2). The kit is connected to a Midi-to-USB midi adapter. It works in general, but there is a latency of approx 500 to 700ms which is far to high. I want to test if a more direct connection between Wii controller and Mac reduces the latency. So i want to connect the Wii Remote via Bluetooth to my Mac.

I tried to run wjoy on Mac OS X 10.13. I start the application from the terminal with "sudo open wjoy". I'm asked for a admin password within the terminal session. After entering that, the wjoy icon appears in the Finder menu and wjoy states "Wjoy says: we are started". But again Mac OS asks for authentication. After entering the password, the wjoy icon disappears from the Finder menu and the application terminates.

This is the terminal log after entering my admin password:
Last login: Sat May 11 21:14:42 on ttys001
/Applications/WJoy.app/Contents/MacOS/WJoy ; exit;
****:~ ***$ /Applications/WJoy.app/Contents/MacOS/WJoy ; exit;
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
[Prozess has ended]
Sometimes the log also says "... closing timed out process (id)"

I just found that if i don't enter the admin password and wait, the wjoy service terminates after 6-7 seconds anyway.
Any idea? Many thanks, your help is much appreciated.
